### ISSUE TITLE
Margins removed in baseline

### ISSUE THREAD

**Issue by pylang** [matplotlib/pytest-mpl#49]
When I run or save a modified README code in Jupyter, I get a different image than the generated baseline.

Example code run in a Jupyter cell:
```python
#%%file test.py
%matplotlib inline

import matplotlib.pyplot as plt
import pytest

@pytest.mark.mpl_image_compare
def test_succeeds():
    fig = plt.figure()
    ax = fig.add_subplot(1,1,1)
    ax.plot([1,2,3])
    fig.savefig("test")
    return fig
```

Running `test_succeeds();` in Jupyter gives resulting image and saved `test.png` **with margins**:

![test](https://cloud.githubusercontent.com/assets/10778668/25150075/b1f6e174-244e-11e7-9dc4-f1d4c126eaff.png)

Next, I generate a `test.py` file by uncommenting the first line, commenting the second line 
and then running the cell 

```python
%%file test.py
#%matplotlib inline

...
```
From a command prompt, in the appropriate directory I run the following to generate the baseline image:

    > py.test --mpl-generate-path=baseline test.py

In the baseline folder, a comparable image called `test_succeeds.png` is created **without margins**:

![test_succeeds](https://cloud.githubusercontent.com/assets/10778668/25150260/58d3509a-244f-11e7-9eeb-eaeca0c5648f.png)

Aside from the difference in sizes, I notice the baseline image uses the classic mpl style.  Perhaps this is the cause for the absence in margins.  I report this as an issue as it is ideal that when testing the baseline image, it should be the same image as shown and saved by Jupyter.

![2017-04-18 16_25_19-scratchpad 5](https://cloud.githubusercontent.com/assets/10778668/25151471/b9c6f5e2-2453-11e7-8440-488148dad383.png)

**Comment by astrofrog** [comment]
Do you have a ``matplotlibrc`` file in ``~/.matplotlib``? I wonder if you have the ``bbox_inches='tight'`` option set by default?

**Comment by pylang** [comment]
No, I don't have any customization in that file.  I tried that latter with `fig.tight_layout()` as well.  I got the same results.  

I  am unable to confirm this with that latest version of matplotlib (due to [this issue](https://github.com/ContinuumIO/anaconda-issues/issues/1068) while upgrading).  I am interested if someone can reproduce these results.

**Comment by ConorMacBride** [comment]
I believe the differences are due to pytest-mpl using the `classic` style by default. [(Matplotlib changed their default style in v2.0)](https://matplotlib.org/stable/users/prev_whats_new/dflt_style_changes.html) To use Matplotlib's current default style in pytest-mpl, test functions should be decorated with `@pytest.mark.mpl_image_compare(style="default")` instead.

---

Most new users today would probably prefer to use Matplotlib's current default style, so maybe we should consider only setting the style if it's explicitly passed to the pytest-mpl decorator? (This is assuming the default style remains stable between major releases of Matplotlib.) This could be a breaking change in pytest-mpl v1.0.0.

**Comment by dopplershift** [comment]
Jupyter by default uses `bbox_inches='tight'` with the matplotlib inline support.
